kamu-core/src/domain/repos/dataset.rs

```diff
@@ -26,6 +27,8 @@ pub trait Dataset: Send + Sync {
     fn as_checkpoint_repo(&self) -> &dyn ObjectRepository;
     fn as_cache_repo(&self) -> &dyn NamedObjectRepository;
 
+    fn base_url(&self) -> &Url;
```

Comment:
Looks like this method is not used any more - probably best to remove it to avoid anyone making assumptions about the specific layout of a dataset under such URL (e.g. `dataset.base_url().join("blocks/abcd")`)